aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,798 @@
+    """
+    Calculates Subnet CIDR address blocks within given VPC CIDR block
+
+    Args:
+        ec2_client: aws boto client
+        cidr_block (str): The IPv4 network of VPC
+        subnet_count (str): The number of additional bits with which to extend the prefix
+
+    Returns:
+        list: The list of subnet CIDR address blocks
+
+    Raises:
+        AddressValueError: If cidr_block provided in invalid for IPv4 address
+    """
+    subnet_range = []
+    try:
+        net = IPv4Network(cidr_block)
+        LOGGER.info(f"cidr block net  get subnets function: {net}")
+        LOGGER.info(f"type : {net}")
+
+        prefix = net.prefixlen + (subnet_count * 2)
+        for subnet in net.subnets(new_prefix=prefix):
+            subnet_range.append(str(subnet))
+        return subnet_range
+    except AddressValueError:
+        LOGGER.error(f"Address is invalid for IPv4: {cidr_block}")
+        raise
+
+
+def get_availability_zones(ec2_client) -> list:
+    """
+    Provides a list of zone IDs in availability zone of current AWS Region
+
+    Args:
+        ec2_client: aws boto client
+
+    Returns:
+        list: The list of Zone IDs
+
+    Raises:
+        ClientError: If error response is provided by an AWS service to Boto3 client’s request.
+    """
+    zones = []
+    try:
+        av_zones = ec2_client.describe_availability_zones()
+        for zone in av_zones.get("AvailabilityZones"):
+            zones.append(zone.get("ZoneId"))
+        LOGGER.info(f" az : {zones}")
+        return zones
+    except ClientError as client_err:
+        LOGGER.error(client_err)
+        raise
+
+
+def create_network_resources(
+    ec2_client, cidr_block, site_name, subnet_count, region_name
+) -> dict:
+    """
+    Creates VPC and other resources(VPC Endpoint, Public and Private Subnets, Route Tables,
+    Route, Nat gateways, Internet Gateways)
+
+    Args:
+        ec2_client: aws boto client
+        cidr_block (str): The IPv4 network for creating VPC
+        site_name (str): The value for 'Name' tag used for all resource
+        subnet_count (str): The number of public and private subnet pairs to create
+
+    Returns:
+        dict: The dictionary contains VPC ID, public and private subnet IDs
+
+    """
+
+    subnet_cidr_blocks = get_subnets_cidr(cidr_block, subnet_count)
+    availability_zones = get_availability_zones(ec2_client)
+
+    # Check the provided number of public and private subnet pairs should not be more than 3
+    # for creating one public and private subnet per AZ
+    if subnet_count > 3:
+        LOGGER.error(
+            f"Number of private and public subnet pairs should be less than 3! \n{subnet_count} is not acceptable!!!",
+        )
+        sys.exit(1)
+
+    vpc_output = create_vpc_and_vpc_endpoint(
+        ec2_client, cidr_block, site_name, region_name
+    )
+    vpc_id = vpc_output["vpc_id"]
+
+    LOGGER.info(f"Creating {subnet_count} Public Subnets...")
```

Comment:
All those loggers should be inside their functions.
`LOGGER.info(f"Creating {subnet_count} Public Subnets...")` should be in `create_public_subnets_for_vpc` and so on

Reply:
done